enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -258,13 +259,16 @@ func (q *PriorityTaskScheduler) EnqueueTaskReservation(ctx context.Context, req
 	return &scpb.EnqueueTaskReservationResponse{}, nil
 }
 
-func propagateExecutionTaskValuesToContext(ctx context.Context, execTask *repb.ExecutionTask) context.Context {
+func (q *PriorityTaskScheduler) propagateExecutionTaskValuesToContext(ctx context.Context, execTask *repb.ExecutionTask) context.Context {
 	ctx = context.WithValue(ctx, "x-buildbuddy-jwt", execTask.GetJwt())
-	rmd := &repb.RequestMetadata{
-		ToolInvocationId: execTask.GetInvocationId(),
+	rmd := execTask.GetRequestMetadata()
+	// TODO(bduffany): remove this fallback once all apps populate request
```

Comment:
I think this can still be nil for the remote bazel synthetic Execute path. 
we recently updated the workflow code to populate the request metadata key, but the remote bazel path has not been updated yet.

Reply:
Thanks -- with that context, I think adding a hard dependency on RequestMetadata seems a bit much. Removed the TODO for now, so we can gracefully handle unexpected cases like this.